From: Jiri Slaby <jslaby@suse.cz>
Subject: Remove headers crude hack

Now, the glibc headers seem to be fixed, so remove this hack which
prevents build on systems with kernel >= 4.20. linux/if_arp.h really
needs to include linux/netdevice.h to have IFNAMSIZ defined. If not,
the build fails miserably:
gcc -DHAVE_CONFIG_H -I. -I../..     -I../../src/include -O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protection -g -Wall -Wshadow -Wpointer-arith -Wwrite-strings -Wstrict-prototypes -c itf.c
In file included from itf.c:17:
/usr/include/linux/if_arp.h:121:16: error: 'IFNAMSIZ' undeclared here (not in a function)
  char  arp_dev[IFNAMSIZ];
                ^~~~~~~~
---
 src/arpd/arp.c |    1 -
 src/arpd/io.c  |    1 -
 src/arpd/itf.c |    1 -
 3 files changed, 3 deletions(-)

--- a/src/arpd/arp.c
+++ b/src/arpd/arp.c
@@ -15,7 +15,6 @@
 #include <sys/types.h>
 #include <sys/socket.h> /* for linux/if_arp.h */
 #include <netinet/in.h> /* for ntohs, etc. */
-#define _LINUX_NETDEVICE_H /* very crude hack for glibc2 */
 #include <linux/types.h>
 #include <linux/if_arp.h>
 #include <linux/if_ether.h>
--- a/src/arpd/io.c
+++ b/src/arpd/io.c
@@ -21,7 +21,6 @@
 #include <atm.h>
 #include <linux/atmclip.h> /* for CLIP_DEFAULT_IDLETIMER */
 #include <linux/atmarp.h>
-#define _LINUX_NETDEVICE_H /* glibc2 */
 #include <linux/types.h>
 #include <linux/if_arp.h>
 
--- a/src/arpd/itf.c
+++ b/src/arpd/itf.c
@@ -12,7 +12,6 @@
 #include <sys/types.h>
 #include <linux/atmclip.h>
 #include <sys/socket.h>
-#define _LINUX_NETDEVICE_H /* glibc2 */
 #include <linux/types.h>
 #include <linux/if_arp.h>
 
